app/helpers/react_on_rails_helper.rb

```diff
@@ -222,31 +226,31 @@ def server_render_js(js_expression, options = {})
 
   private
 
-  # prepend the rails_context if not yet applied
-  def prepend_render_rails_context(render_value)
+  # add the rails_context if not yet applied
+  def add_render_rails_context(render_value)
     return render_value if @rendered_rails_context
 
-    data = {
-      rails_context: rails_context(server_side: false)
-    }
-
     @rendered_rails_context = true
 
-    rails_context_content = content_tag(:div,
-                                        "",
+    rails_context_content = content_tag(:script,
                                         id: "js-react-on-rails-context",
-                                        style: ReactOnRails.configuration.skip_display_none ? nil : "display:none",
-                                        data: data)
-    "#{rails_context_content}\n#{render_value}".html_safe
+                                        style: nil) do
+      "var #{'js-react-on-rails-context'.tr('-', '_')} = #{Yajl.dump(rails_context(server_side: false))};".html_safe
+    end
+    "#{render_value}\n#{rails_context_content}".html_safe
   end
 
   def render_redux_store_data(redux_store_data)
-    result = content_tag(:div,
+    result = content_tag(:script,
                          "",
                          class: "js-react-on-rails-store",
-                         style: ReactOnRails.configuration.skip_display_none ? nil : "display:none",
-                         data: redux_store_data)
-    prepend_render_rails_context(result)
+                         style: nil,
+                         data: redux_store_data) do
+      # redux_store_data
+      "var #{redux_store_data[:store_name].tr('-', '_')} = #{Yajl.dump(redux_store_data[:props])};".html_safe
+    end
```

Comment:
Generating a JavaScript string from Ruby is a pretty bad idea in my experience. For one thing the generated identifier is in the global namespace, but it’s also potentially very fragile.

My personal preference for getting values from Ruby into JS is to provide them as `data-` attributes on some element, then pull them out with JS… but that seems to be pretty much what you were doing already. Is there some reason why you wanted to use a `<script>` tag rather than a `<div>`?